### PR TITLE
[SYCL] Use clang++ for LIT tests, link sycl.lib with -fsycl on Windows

### PR DESCRIPTION
**Pull request by v-klochkov**

### CODE REVIEW

File: sycl/test/basic_tests/device.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: %clang -std=c++11 -g %s -o %t.out -lstdc++ -lOpenCL -lsycl
+// RUN: %clangxx -g %s -o %t.out -lOpenCL -lsycl
```
Comment (bader) [inline sycl/test/basic_tests/device.cpp]: I suggest removing `-g` too.

File: sycl/test/basic_tests/event.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// RUN: %clang -std=c++11 -g %s -o %t.out -lstdc++ -lOpenCL -lsycl
+// RUN: %clangxx -fsycl -g %s -o %t.out -lOpenCL
```
Comment (bader) [inline sycl/test/basic_tests/event.cpp]: I suggest removing -g too.

File: sycl/test/basic_tests/macros.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// RUN: %clang -std=c++11 -g %s -o %t.out -lstdc++ -lOpenCL -lsycl
+// RUN: %clangxx -g %s -o %t.out -lOpenCL -lsycl
```
Comment (bader) [inline sycl/test/basic_tests/macros.cpp]: I suggest removing -g too.

File: sycl/test/basic_tests/device.cpp (hunk 1 above)
Comment (v-klochkov) [reply sycl/test/basic_tests/device.cpp]: Ok, removed -g

File: sycl/test/basic_tests/event.cpp (hunk 2 above)
Comment (v-klochkov) [reply sycl/test/basic_tests/event.cpp]: Ok, removed -g

**Comment by v-klochkov** [comment]
During jumps from Windows to Linux and back I lost a minor fix to context.cpp test. Fixed it now.

**Pull request merged**